alibi_detect/cd/model_uncertainty.py

```diff
@@ -27,6 +29,7 @@ def __init__(
             device: Optional[str] = None,
             tokenizer: Optional[Callable] = None,
             max_len: Optional[int] = None,
+            input_shape: Optional[tuple] = None,
```

Comment:
This change seems unrelated to this PR?

Reply:
The `input_shape` info is used when saving TensorFlow models. It was missing from detectors that didn't have save functionality with the legacy `save_detector`, so has now been added since all drift detectors are now "saveable".

Reply:
Does it mean the detectors are only saveable if the correct `input_shape` is passed by the user? Is this documented?

Reply:
The detector would be saveable without, but only if it didn't have a `model` defined as part of its `preprocess_fn`. It isn't documented. I'm now actually leaning towards introducing a new top-level drift detector base class to handle args/kwargs shared across all detectors, in which case we could add including `input_shape` (and `data_type` etc) to this issue: https://github.com/SeldonIO/alibi-detect/issues/512.

I've already done something similar for the pydantic models, where we have a top-level `DriftDetectorConfig`...